src/libcollections/slice.rs

```diff
@@ -343,15 +343,17 @@ impl<T> [T] {
         core_slice::SliceExt::last_mut(self)
     }
 
-    /// Returns the element of a slice at the given index, or `None` if the
-    /// index is out of bounds.
+    /// Returns the element or subslice of a slice at the given position or
```

Comment:
It might be more clear to spell out the overloading in more detail here -- that is, to say something like:

- If given a position, returns the element at that position of `None` if out of bounds.
- If given a range, returns the subslice corresponding to that range, or `None` if out of bounds.

And similarly elsewhere.

Otherwise r=me once travis is happy.